velox/row/tests/CompactRowTest.cpp

```diff
@@ -49,35 +49,57 @@ class CompactRowTest : public ::testing::Test, public VectorTestBase {
 
     auto rowType = asRowType(data->type());
     auto numRows = data->size();
+    std::vector<size_t> rowSize(numRows);
+    std::vector<size_t> offsets(numRows);
 
     CompactRow row(data);
 
     size_t totalSize = 0;
     if (auto fixedRowSize = CompactRow::fixedRowSize(rowType)) {
       totalSize = fixedRowSize.value() * numRows;
+      for (auto i = 0; i < numRows; ++i) {
+        rowSize[i] = fixedRowSize.value();
+        offsets[i] = fixedRowSize.value() * i;
+      }
     } else {
       for (auto i = 0; i < numRows; ++i) {
-        totalSize += row.rowSize(i);
+        rowSize[i] = row.rowSize(i);
+        offsets[i] = totalSize;
+        totalSize += rowSize[i];
       }
     }
 
-    std::vector<std::string_view> serialized;
-
     BufferPtr buffer = AlignedBuffer::allocate<char>(totalSize, pool(), 0);
     auto* rawBuffer = buffer->asMutable<char>();
-    size_t offset = 0;
-    for (auto i = 0; i < numRows; ++i) {
-      auto size = row.serialize(i, rawBuffer + offset);
-      serialized.push_back(std::string_view(rawBuffer + offset, size));
-      offset += size;
+    {
+      size_t offset = 0;
+      std::vector<std::string_view> serialized;
+      for (auto i = 0; i < numRows; ++i) {
+        auto size = row.serialize(i, rawBuffer + offset);
```

Comment:
@rui-mo The previous API is still tested here. The two code blocks on line 74 and line 93 are doing the same thing with the prev/new APIs. If we remove the previous API, we should remove this test block as well.